spec.bs

```diff
@@ -1211,18 +1267,26 @@ Several APIs specific to fenced frames are defined on the {{Fence}} interface.
   };
 
   dictionary FenceEvent {
-    required DOMString eventType;
-    required DOMString eventData;
-    required sequence&lt;FenceReportingDestination&gt; destination;
+    // This dictionary has two mutually exclusive modes:
```

Comment:
I can't remember, is there a technical reason why we can't express this as a [union type](https://webidl.spec.whatwg.org/#dfn-union-type), and instead have to express this as a dictionary with these two implicit modes? If there is we should document it here, and if there is not we should try and follow-up with a web-compatible change to make this more idiomatic and less error-prone.

Reply:
It was due to distinguishability. The issue isn't as clear in the spec because reportEvent for component ads hasn't been spec'd yet, which makes some of the required fields optional.